[IMP] Grap Cooperative : activity became res company

quentinDupont opened the pull request:
- Remove grap activity model → became res.compay
- Remove models : grap_activity, grap_activity_people, grap_member, grap_activity_type
- kanban view imitate old kanban view
- new form view
- new fields on res.company
- Migration for keeping datas between 12.0.1 and 12.0.2
 
 Utile de merger cette PR avec https://github.com/odoo-cae/odoo-addons-cae/pull/33
 
![image](https://user-images.githubusercontent.com/9005817/121670766-175ca580-caae-11eb-8510-b7ad880b0085.png)

Grap : apps/deck/#/board/144/card/1267

quentinDupont committed:
WIP - grap activity became res company

legalsylvain committed:
[FIX] dependency + first migration script

legalsylvain committed:
[FIX] avec le dossier de migraiton, c'est mieux

legalsylvain commented on grap_cooperative/models/res_company.py:
```diff
@@ -0,0 +1,116 @@
+        string="Accounting closure",
+    )
+
+    is_using_Odoo = fields.Boolean(
```
yop ! plutot en minuscule les noms par convention. ``is_using_odoo``

quentinDupont committed:
[FIX] and [IMP]

quentinDupont committed:
FR Translation

quentinDupont committed:
FIX travis

quentinDupont committed:
fix travais : add dependency

quentinDupont committed:
FIX views + demo datas

quentinDupont committed:
IMP scalability, view

legalsylvain commented on grap_cooperative/demo/grap_people.xml:
```diff
@@ -20,7 +24,11 @@ License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
         <field name="last_name">DUPONT</field>
+        <field name="name">DUPONT Quentin</field>
         <field name="birthdate">1972-06-19/</field>
+        <field name="street">Street Ambroise Croizat</field>
```
héhé.

legalsylvain commented on grap_cooperative/migrations/12.0.2.0.0/pre-migration.py:
```diff
@@ -0,0 +1,75 @@
+            hr_interlocutor_service_id = ga.hr_interlocutor_id,
+            attendant_interlocutor_service_id =ga.attendant_interlocutor_id
+        FROM grap_activity ga
+        WHERE res_company.name ILIKE ('%' || ga.activity_name || '%');
```
``WHERE res_company.code = ga.code`` serait peut-être plus prédictif, et moins sujet aux aléat de fautes d'ortographe, ou d'accent ?

quentinDupont replied:
au top , bien plus simple en effet !

legalsylvain commented on grap_cooperative/models/grap_people.py:
```diff
@@ -74,24 +87,37 @@ def _compute_is_birthday(self):
+    # Set people name with first and last name
+    # Optimize for creating people directly from company form
+    @api.onchange("first_name", "last_name")
+    def _handle_name(self):
```
pas compris. on peut en causer ?

legalsylvain commented on grap_cooperative/models/grap_people.py:
```diff
@@ -74,24 +87,37 @@ def _compute_is_birthday(self):
 
     def unlink(self):
-        members = self.mapped("grap_member_id")
+        self.write({"company_id": False})
```
pourquoi ?

quentinDupont replied:
grap_member ça n'existe plus du coup
c'est direct grap.people → res.company

Et du coup j'ai changé de unlink() à detach_people_from_company() car le but n'est plus de supprimer mais juste de détacher le lien entre people et company (je me dis qu'on a envie de garder la fiche des grap.people même si la personne part de l'activité (au cas où elle revient dans une autre acti par exemple))

quentinDupont replied:
> la migration des liens grap.people -> res.company est cassé. Cf 3PP, BSG.

du coup j'ai plus la base de test pour voir ça hmm. P-e que ça passera mieux avec la modif sur le script de migration

legalsylvain replied:
:+1:

legalsylvain commented on grap_cooperative/models/res_company.py:
```diff
@@ -0,0 +1,135 @@
+        ("obsolete", "Exited"),
+    ]
+
+    display_name = fields.Char(
```
ce champs existe déjà dans Odoo (res.company.display_name). peut-etre que pour éviter les conflits, ça serait chouette de mettre plutôt un champ ``clean_name``.

quentinDupont replied:
gravos daccoros

legalsylvain commented on grap_cooperative/models/res_company.py:
```diff
@@ -0,0 +1,135 @@
+            if activity.city:
+                activity.clean_address += activity.city.upper()
+
+    # manager_ids = fields.Many2many(
```
ligne 111 à 135 : ça a vocation a être décommenté ? ou on peut shooter ?

quentinDupont replied:
a shooté

legalsylvain reviewed (commented):
qq remarques.

sinon, de façon plus général : 
- dans le menu, vu qu'il y a moins d'item, je mettrais bien au niveau 1 les éléments "Collèges" et mandats" pour qu'il soit plus visible.
- le modèle ``grap.category`` est supprimé avec cette PR. est-ce qu'on ne créerait pas un élément de menu avec une vue sur ``res.company.category`` ? (qui peut ou prou est la même chose).

legalsylvain reviewed (changes requested):
la migration des liens grap.people -> res.company est cassé. Cf 3PP, BSG.

quentinDupont committed:
Change menus and add menu company_category

quentinDupont commented:
> qq remarques.
> 
> sinon, de façon plus général :
> 
>     * dans le menu, vu qu'il y a moins d'item, je mettrais bien au niveau 1 les éléments "Collèges" et mandats" pour qu'il soit plus visible.
> 
>     * le modèle `grap.category` est supprimé avec cette PR. est-ce qu'on ne créerait pas un élément de menu avec une vue sur `res.company.category` ? (qui peut ou prou est la même chose).

Bonnes idées !
J'ai fait les deux :heavy_check_mark:

quentinDupont committed:
Change unlink → detach people from company

quentinDupont committed:
Traduction et clean_name storé

quentinDupont committed:
change migration

quentinDupont commented:
> la migration des liens grap.people -> res.company est cassé. Cf 3PP, BSG.

du coup j'ai plus la base de test pour voir ça hmm. P-e que ça passera mieux avec la modif sur le script de migration

legalsylvain committed:
[FIX] rename views.

legalsylvain committed:
[REF] grap_cooperative : simplifiy activity state. (replace by company boolean 'is_displayed_in_directory')
[MIG] initialize res_company.is_displayed_in_directory  from grap_activity.state
[FIX] fix migration. (link between grap_people and res_company). Base script on code.
[REF] temporary remove accounting_closing_date.
[ADD] correct security access for the crew members
[UPD] translation
[FIX] set correct context in the company_id field, to display correct form view
[REF] remove useless code
[ADD] private phone on people kanban view

legalsylvain commented:
Yop. J'ai fait un gros commit là : https://github.com/grap/grap-odoo-custom/pull/174/commits/5da5b362bf04d9bcaba15f7a61ae94bf8d170634

- [MIG] initialize res_company.is_displayed_in_directory  from grap_activity.state
- [FIX] fix migration. (link between grap_people and res_company). Base script on code.
- [REF] temporary remove accounting_closing_date.
- [ADD] correct security access for the crew members
- [UPD] translation
- [FIX] set correct context in the company_id field, to display correct form view
- [REF] remove useless code
- [ADD] private phone on people kanban view

Si tu as des questions / demandes de précision sur certains trucs, tu n'hésites pas.

legalsylvain commented:
MeP du 28/07/2021.

/ocabot merge patch
